### PR TITLE
Robbie issue 1 continued

### PR DESCRIPTION
**Pull request by robbie-smith**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -13,4 +13,35 @@ def add_item(item_id)
     contents[item_id.to_s] ||= 0
     contents[item_id.to_s] += 1
   end
+
+  def price_and_quantity
+    found_items = []
+    contents.each_pair do |id,quantity|
```
Comment (CPowell23) [inline app/models/cart.rb:19]: Consider refactoring to a map. ex: contents.map do |(id , quantity)|...

File: app/models/cart.rb [hunk 2]
```diff
@@ -13,4 +13,35 @@ def add_item(item_id)
...
+    found_items
+  end
+
+  def items
+    Item.find(contents.keys)
+  end
+
+  def items_with_quantities
+    item_quantity = {}
+    contents.map do |id, quantity|
```
Comment (CPowell23) [inline app/models/cart.rb:31]: Consider converting .map to reduce

File: app/models/cart.rb [hunk 3]
```diff
@@ -13,4 +13,35 @@ def add_item(item_id)
...
+
+  def total_price_in_cents
+    price_and_quantity.reduce(0) do |memo, prices|
+      memo += prices[0] * prices[1]
+      memo
+    end
+  end
+
+  def total_price_in_dollars
+    total_price_in_cents / 100.to_f
```
Comment (CPowell23) [inline app/models/cart.rb:45]: Consider a helper method for price in dollars because we're using this in multiple models

File: app/models/cart.rb [hunk 4]
```diff
@@ -13,4 +13,35 @@ def add_item(item_id)
...
+
+  def price_and_quantity
+    found_items = []
+    contents.each_pair do |id,quantity|
+      found_items <<[ Item.find(id).price_in_cents, quantity ]
+    end
+    found_items
+  end
+
+  def items
```
Comment (CPowell23) [inline app/models/cart.rb:25]: Do we still need this?

File: spec/features/cart/user_can_view_items_in_cart_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+feature "When a customer visits the home page and wants to view their cart" do
+
+  let!(:pants) { create(:item, title: "Rustic Wooden Pants", price_in_cents: 11_11) }
+  let!(:computer) { create(:item, title: "Synergistic Silk Computer", price_in_cents: 22_22) }
+  let!(:spoon) { create(:item, title: "Four Dollar Wooden Spoon", price_in_cents: 33_33) }
+
+  before do
```
Comment (CPowell23) [inline spec/features/cart/user_can_view_items_in_cart_spec.rb:9]: Can we also add a sad path test for what happens when they try to view the cart when the cart is empty?
